litmus-portal/backend/workflow-agent/pkg/types/event.go

```diff
@@ -0,0 +1,38 @@
+package types
+
+// workflow data
+type WorkflowEvent struct {
+	EventType         string          `json:"event_type"`
+	UID               string          `json:"uid"`
+	Namespace         string          `json:"namespace"`
+	Name              string          `json:"name"`
+	CreationTimestamp string          `json:"creationTimestamp"`
+	Phase             string          `json:"phase"`
+	StartedAt         string          `json:"startedAt"`
+	FinishedAt        string          `json:"finishedAt"`
+	Nodes             map[string]Node `json:"nodes"`
+}
+
+// each node/step data
+type Node struct {
+	Name       string     `json:"name"`
+	Phase      string     `json:"phase"`
+	StartedAt  string     `json:"startedAt"`
+	FinishedAt string     `json:"finishedAt"`
+	Children   []string   `json:"children"`
+	Type       string     `json:"type"`
+	ChaosExp   *ChaosData `json:"chaosData,omitempty"`
+}
+
+// chaos data
+type ChaosData struct {
```

Comment:
Can we reuse this struct from chaos-operator? Any struct change to chaos operator will not reflect here

Reply:
We do not use all the items from the engine, this needs to be a custom one.